src/formatting/items.rs

```diff
@@ -235,18 +235,18 @@ pub(crate) struct FnSig<'a> {
 
 impl<'a> FnSig<'a> {
     pub(crate) fn new(
```

Comment:
I think it may be easier to use `from_method_sig`, which allows for dropping this method entirely. See  https://github.com/rust-lang/rustfmt/pull/4290/files

Reply:
Good point! I like this idea better than mine. Thank you for creating the PR, and no worries about the duplicated work. GitHub UI is laggy these days.

Reply:
Related aside, we should probably rename `from_method_sig` to `from_fn_sig`. Believe that's one that we missed when incorporating the upstream rustc Method -> Fn renaming